lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -115,7 +115,9 @@ def bulk(actions)
                     action.map {|line| LogStash::Json.dump(line)}.join("\n") :
                     LogStash::Json.dump(action)
         as_json << "\n"
-        bulk_responses << bulk_send(body_stream) if (body_stream.size + as_json.bytesize) > TARGET_BULK_BYTES
+	if (body_stream.size + as_json.bytesize) > TARGET_BULK_BYTES
```

Comment:
nitpick: since the only statement in this `if`...`end` block has a tail-conditional, it's easy to lose track of what we're doing; perhaps the following will read more clearly, and as a bonus will short-circuit out of doing the will-we-overflow calculation if our `body_stream` is empty:

~~~ ruby
        if body_stream.size > 0 && (body_stream.size + as_json.bytesize) > TARGET_BULK_BYTES
          bulk_responses << bulk_send(body_stream)
        end
~~~